src/main/java/net/kyori/text/KeybindComponent.java

```diff
@@ -51,15 +80,104 @@ public String keybind() {
+  }
+
+  @Nonnull
+  @Override
+  public KeybindComponent insertion(@Nullable String insertion) {
+    return new KeybindComponent(this.children, this.color, this.obfuscated, this.bold, this.strikethrough, this.underlined, this.italic, this.clickEvent, this.hoverEvent, insertion, this.keybind);
+  }
+
+  @Nonnull
+  @Override
+  public KeybindComponent mergeStyle(@Nonnull final Component that) {
+    return new KeybindComponent(this.children, that.color(), that.decoration(TextDecoration.OBFUSCATED), that.decoration(TextDecoration.BOLD), that.decoration(TextDecoration.STRIKETHROUGH), that.decoration(TextDecoration.UNDERLINE), that.decoration(TextDecoration.ITALIC), that.clickEvent(), that.hoverEvent(), that.insertion(), this.keybind);
+  }
+
+  @Nonnull
+  @Override
+  public KeybindComponent mergeColor(@Nonnull Component that) {
+    return new KeybindComponent(this.children, that.color(), this.obfuscated, this.bold, this.strikethrough, this.underlined, this.italic, this.clickEvent, this.hoverEvent, this.insertion, this.keybind);
+  }
+
+  @Nonnull
+  @Override
+  public KeybindComponent mergeDecorations(@Nonnull Component that) {
+    final TextDecoration.State obfuscated = that.decoration(TextDecoration.OBFUSCATED) != TextDecoration.State.NOT_SET ? that.decoration(TextDecoration.OBFUSCATED) : this.obfuscated;
+    final TextDecoration.State bold = that.decoration(TextDecoration.BOLD) != TextDecoration.State.NOT_SET ? that.decoration(TextDecoration.BOLD) : this.bold;
+    final TextDecoration.State strikethrough = that.decoration(TextDecoration.STRIKETHROUGH) != TextDecoration.State.NOT_SET ? that.decoration(TextDecoration.STRIKETHROUGH) : this.strikethrough;
+    final TextDecoration.State underlined = that.decoration(TextDecoration.UNDERLINE) != TextDecoration.State.NOT_SET ? that.decoration(TextDecoration.UNDERLINE) : this.underlined;
+    final TextDecoration.State italic = that.decoration(TextDecoration.ITALIC) != TextDecoration.State.NOT_SET ? that.decoration(TextDecoration.ITALIC) : this.italic;
+    return new KeybindComponent(this.children, this.color, obfuscated, bold, strikethrough, underlined, italic, this.clickEvent, this.hoverEvent, this.insertion, this.keybind);
+  }
+
+  @Nonnull
+  @Override
+  public KeybindComponent mergeEvents(@Nonnull Component that) {
+    return new KeybindComponent(this.children, this.color, this.obfuscated, this.bold, this.strikethrough, this.underlined, this.italic, that.clickEvent(), that.hoverEvent(), this.insertion, this.keybind);
+  }
+
+  @Nonnull
+  @Override
+  public KeybindComponent resetStyle() {
+    return new KeybindComponent(this.children, null, TextDecoration.State.NOT_SET, TextDecoration.State.NOT_SET, TextDecoration.State.NOT_SET, TextDecoration.State.NOT_SET, TextDecoration.State.NOT_SET, null, null, null, this.keybind);
+  }
+
+  @Nonnull
+  @Override
+  public Component copy() {
```

Comment:
Not sure if it might be worth Specifying the current `Component` subclass as a return type of `copy`. I don't believe there is much to gain currently, but it could make a difference in the future.